framework/src/utils/MatrixTools.C

```diff
@@ -43,7 +45,14 @@ void
 inverse(std::vector<PetscScalar> & A, unsigned int n)
 {
   mooseAssert(n >= 1, "MatrixTools::inverse - n (leading dimension) needs to be positive");
-  mooseAssert(n == static_cast<int>(n), "MatrixTools::inverse - n (leading dimension) too large");
+
+#ifndef NDEBG
+  constexpr auto N = CHAR_BIT * sizeof(unsigned int);
+
+  std::bitset<N> most_significant_bit_test(n);
```

Comment:
If you are worried about the signedness warning in the asset then just use
std limits!

On Mon, Jun 13, 2016, 4:20 PM John W. Peterson <notifications@github.com>
wrote:

> In framework/src/utils/MatrixTools.C
> <https://github.com/idaholab/moose/pull/7183#discussion_r66886044>:
>
> > @@ -43,7 +45,14 @@ void
> >  inverse(std::vector<PetscScalar> & A, unsigned int n)
> >  {
> >    mooseAssert(n >= 1, "MatrixTools::inverse - n (leading dimension) needs to be positive");
> > -  mooseAssert(n == static_cast<int>(n), "MatrixTools::inverse - n (leading dimension) too large");
> > +
> > +#ifndef NDEBG
> > +  constexpr auto N = CHAR_BIT * sizeof(unsigned int);
> > +
> > +  std::bitset<N> most_significant_bit_test(n);
>
> What the heck is all this std::bitset stuff?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7183/files/3d359ba5b9ba3b80b951b352e9d2cd9f7a20a6b3#r66886044>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMWPkc7k7yUNCKlkD8WNgjc4KPjafg5ks5qLeW1gaJpZM4I0vZG>
> .
>